file name encryption/AES-SIV-512-B64URL.md

```diff
@@ -79,6 +79,66 @@ let dirPath = vaultRoot + '/d/' + dirIdString[0..2] + '/' + dirIdString[2..32]
 
 Regardless of the hierarchy of cleartext paths, ciphertext directories are always stored in a flattened structure. All directories will therefore effectively be siblings (or cousins, to be precise).
 
+```mermaid
+---
+title: Overview Mapping Directory IDs to Paths, Encryption of Directory Names and Directory Metadata
+---
+flowchart TD
+   decision{root?}
+   decision -->|n| latestSeed
+   decision -->|n| csprng32
+   decision -->|y| initialSeed
+   latestSeed --> directorySeed
+   csprng32 --> dirId
+   csprng32{{"csprng(32)"}}
+   initialSeed --> directorySeed
+   initialSeed -->|seed:| kdfRootDirId
+   kdfRootDirId{{"kdf(seed,32,'rootDirId')"}}
+   kdfRootDirId --> dirId
+   directorySeed -->|seed:| kdfSiv
+   kdfSiv{{"kdf(seed,64,'siv')"}}
+   kdfSiv --> sivKey
+   directorySeed -->|seed:| kdfHmac
+   kdfHmac{{"kdf(seed,32,'hmac')"}}
+   kdfHmac --> hmacKey
+   hmacKey -->|key:| hmacSha256
+   hmacSha256{{hmacSha256}}
+   hmacSha256 --> dirIdHash
+   dirIdHash --> truncate
+   truncate{{"_[0..20]"}}
+   truncate --> base32
+   base32{{base32}}
+   base32 --> dirIdString
+   dirIdString --> head
+   head{{"_[0..2]"}}
+   dirIdString --> tail
+   tail{{"_[2..32]"}}
+   head -->|$0:| pattern
+   tail -->|$1:| pattern
+   pattern{{"'d/'$0'/'$1"}}
+   pattern --> dirPath
+   sivKey -->|sivKey:| aesSiv
+   aesSiv{{aesSiv}}
+   parentDirId -->|ad:| aesSiv
+   clearTextName -->|secret:| aesSiv
+   aesSiv --> base64Url
+   base64Url{{base64url}}
+   base64Url --> join
+   uvf --> join
+   uvf["'.uvf'"]
+   join{{"join"}}
+   join --> ciphertextName
+   directorySeed --> concat
+   dirId --> concat
+   concat{{concat}}
+   concat -->|cleartextBlocks:| fileContentEncryption
+   fileContentEncryption{{file content encryption}}
+   directorySeed -->|seed:| fileContentEncryption
+   fileContentEncryption --> dirUvf
+   dirUvf["directory metadata dir.uvf content"]
+```
+
+
```

Comment:
I like the idea, but let's break this into two charts to make it more digestible:

1. decide which dirId and seed is used
2. derive dir path and encrypt node names

Reply:
Done: 92dbd884e51ba0dac60d8b024af0f6ff702a6bed